sdks/python/apache_beam/io/gcp/experimental/spannerio.py

```diff
@@ -109,20 +111,74 @@
 
 ReadFromSpanner takes this transform in the constructor and pass this to the
 read pipeline as the singleton side input.
+
+Writing Data to Cloud Spanner.
+
+The WriteToSpanner transform writes to Cloud Spanner by executing a
+collection a input rows (WriteMutation). The mutations are grouped into
+batches for efficiency.
+
+WriteToSpanner transform relies on the WriteMutation objects which is exposed
+by the SpannerIO API. WriteMutation have five static methods (insert, update,
+insert_or_update, replace, delete). These methods returns the instance of the
+_Mutator object which contains the mutation type and the Spanner Mutation
+object. For more details, review the docs of the class SpannerIO.WriteMutation.
+For example:::
+
+  mutations = [
+                WriteMutation.insert(table='user', columns=('name', 'email'),
+                values=[('sara'. 'sara@dev.com')])
+              ]
+  _ = (p
+       | beam.Create(mutations)
+       | WriteToSpanner(
+          project_id=SPANNER_PROJECT_ID,
+          instance_id=SPANNER_INSTANCE_ID,
+          database_id=SPANNER_DATABASE_NAME)
+        )
+
+You can also create WriteMutation via calling its constructor. For example:::
+
+  mutations = [
+      WriteMutation(insert='users', columns=('name', 'email'),
+                    values=[('sara", 'sara@example.com')])
+  ]
+
+For more information, review the docs available on WriteMutation class.
+
+WriteToSpanner transform also takes 'max_batch_size_bytes' param which is set
+to 1MB (1048576 bytes) by default. This parameter used to reduce the number of
```

Comment:
There is one other batching parameter which is important -- the maximum number of cells being mutated. Spanner has a hard 20K limit here, so a batch must have less than 20K mutated cells, including cells being mutated in indexes. 

Java version sets this to 5K by default. 

A third parameter max_number_rows was also added recently to java, limiting the total number of rows in a batch.

Reply:
For my understanding, `maximum_number_cells` would the (total number of columns * total number of rows)

For Example:
```
WriteMutation.insert("roles", ("key", "rolename"), [('abc1', "test-1"), ('abc2', "test-2"), ('abc3', "test-3")])
```
in this case... the max_number_cells would be `2 * 3 = 6`


And for the max_rows_number will be 2 in the below case.

```
MutationGroup([
            WriteMutation.insert("roles", ("key", "rolename"),
                                 [('abc1', "test1")]),
            WriteMutation.insert("roles", ("key", "rolename"),
                                 [('abc2', "test2")])
        ])
```
Please corrent me if I am mistaken!

Reply:
No, you are correct.

Reply:
Also, given that in this transform the batches are not pre-sorted, I would make the defaults a lot smaller than the Java equivalent: say max 500 cells per batch, and max 50 rows.

Reply:
Thanks. I'll update the code!